feat(dropdown): display a dialog with filtering for > 100 choices

mturoci opened the pull request:
I am not sure I understand the meaning of `caption` in `ui.choice` as proposed in #153, so not implemented yet. What is the difference between `caption` and `label`? 

Closes #153

shanaka-rajitha commented:
@lo5 @mturoci  I have added few tweaks to the existing dropdown component.

**Variations on default and active state of dropdown**
![Dropdown list](https://user-images.githubusercontent.com/12246331/117424924-ebdf1c00-af3f-11eb-8d43-9fb14ea1ef61.png)

**Default dialog with filtering using checkbox**

- Since user may select any number of choices we need to use checkbox, radio buttons are used when user must select exactly one choice.

![Modal - Dropdown list](https://user-images.githubusercontent.com/12246331/117424940-ef72a300-af3f-11eb-867b-1b0b7efa9197.png)


**Default dialog with filtering using search option**

<img width="778" alt="Modal - Dropdown list-2" src="https://user-images.githubusercontent.com/12246331/117426201-4167f880-af41-11eb-852a-5596e9de3a56.png">

mturoci commented:
I have updated design according to Fluent:

https://user-images.githubusercontent.com/64769322/118106622-13216780-b3de-11eb-9efd-53c89f79df7d.mov

* used Fluent Searchbar instead of custom one to keep consistency
* used Detaillist with always visible checkboxes instead of regular checkboxes. The technical reason is that Detaillist is virtualized and checkboxes are not. Considering we already had cases where people wanted to filter large data using `ui.picker` and faced perf issues, this seem like a good quick solution. Other alternative would be to write virtualization ourselves, but I think it would be a waste of time - cc @lo5 
* Added select / unselect all as proposed in design, but used styles we use for other components currently to keep consistency.

Needs discussion:
* The current UX clears current selection when filtering. This is caused because Fluent tightly couples items in Detaillist with selection items for some reason. The question is whether this is good UX as it is or we should keep the selection throughout filtering - cc @lo5 @shanaka-rajitha
